httpclient/useragent.go

```diff
@@ -38,3 +40,132 @@ func (rt *userAgentRoundTripper) RoundTrip(req *http.Request) (*http.Response, e
 	}
 	return rt.inner.RoundTrip(req)
 }
+
+func UserAgent(version string) *userAgent {
+	return newUserAgent([]*UserAgentProduct{
+		{"HashiCorp", "1.0", ""},
+		{"Terraform", version, "+https://www.terraform.io"},
+	})
+}
+
+type UserAgentProduct struct {
+	Name    string
+	Version string
+	Comment string
+}
+
+func (uap *UserAgentProduct) String() string {
+	var b strings.Builder
+	b.WriteString(uap.Name)
```

Comment:
is it possible to just have a comment and no name and version?

Reply:
Theoretically yes, nothing prevents you from either not setting `Name` nor `Version`, but I think that would not be a conventional/valid UA string per [RFC7231](https://tools.ietf.org/html/rfc7231#section-5.5.3) and `ParseUserAgentString` would not accept it.

**Should we prevent invalid UA format?**
The problem with any such validation is that it's likely to hurt usability and effectivity would be questionable. I can't imagine adding `error` to `String()`, so I reckon we would have to just log `WARN` or `ERROR` and that's easily ignorable.

**Should we make it easier to use invalid UA format?**
e.g. trim the space in front of the comment? I don't think so. If you pass something invalid, the output should be invalid, basically telling you not to do that. I don't think we should attempt to fix what is wrong.

Reply:
As discussed, `(comment)` is actually a valid UA string. Implementation fixed in ba9d562